src/rgw/rgw_acl_swift.cc

```diff
@@ -324,16 +324,16 @@ int create_account_policy(const DoutPrefixProvider* dpp,
   return 0;
 }
 
-} // namespace rgw::swift
-
-boost::optional<std::string> RGWAccessControlPolicy_SWIFTAcct::to_str() const
+std::string format_account_acl(const RGWAccessControlPolicy& policy)
```

Comment:
i dont see the value in the change here?
also, isn;t it better to keep the `std::optional` then to use the empty string?

Reply:
making this a free function was a prerequisite to removing the `RGWAccessControlPolicy_SWIFTAcct` class

regarding std::optional, the only empty-optional case was here:
```c++
  /* If there is no grant to serialize, let's exit earlier to not return
   * an empty JSON object which brakes the functional tests of Swift. */
  if (admin.empty() && readwrite.empty() && readonly.empty()) {
```
in all other cases, we return a non-empty string. since we don't need to represent an empty optional and an empty string as different states, it's simpler to just rely on the empty-string state. this behavior is documented in the header as:
```c++
/// Returns an empty string if there are no admin/read-write/read-only entries.
```

Reply:
that said, 'better' is subjective - what do you think?

Reply:
using `std::optional` does not require a comment, while usign an empty string does require one.
to me, this is an indication that "optional" is better. but if the only place it is used already have a bunch of "empty()" checks we can keep the string